src/devices_models/devices/common/add_to_expression.jl

```diff
@@ -196,6 +196,34 @@ function add_to_expression!(
     return
 end
 
+function add_to_expression!(
+    container::OptimizationContainer,
+    ::Type{T},
+    ::Type{U},
+    devices::IS.FlattenIteratorWrapper{V},
+    ::DeviceModel{V, W},
+    ::Type{X},
+) where {
+    T <: SystemBalanceExpressions,
+    U <: OnVariable,
+    V <: PSY.ThermalGen,
+    W <: AbstractDeviceFormulation,
+    X <: PM.AbstractPowerModel,
+}
+    variable = get_variable(container, U(), V)
+    expression = get_expression(container, T(), X)
+    for d in devices, t in get_time_steps(container)
+        name = PSY.get_name(d)
+        bus_number = PSY.get_number(PSY.get_bus(d))
+        _add_to_jump_expression!(
+            expression[bus_number, t],
+            variable[name, t],
+            get_variable_multiplier(U(), d, W()),
```

Comment:
@jd-lara I had to add few method like this one for Compact thermal formulation, because here to know the multiplier for On we need the device to for retrieving the Pmin. 
The other methods were using this 
https://github.com/NREL-SIIP/PowerSimulations.jl/blob/9d4f952a25d27b0366cc3e659f894fb6e0443e78/src/devices_models/devices/thermal_generation.jl#L9
but this is the one we need here 
https://github.com/NREL-SIIP/PowerSimulations.jl/blob/9d4f952a25d27b0366cc3e659f894fb6e0443e78/src/devices_models/devices/thermal_generation.jl#L10